### ISSUE TITLE
IServicePartitionResolver is not unit testable

### ISSUE THREAD

**Issue by rtanase** [Azure/service-fabric-services-and-actors-dotnet#85]
It looks like the ResolvedServicePartition doesn't have  a public constructor, so we can't stub the ResolveAsync methods.

**Comment by amanbha** [comment]
@rtanase  Thanks for reporting, I will look into this further to see whats the best way to allow it, as giving it a public constructor will not help as it creates other members from the native objects, probably removing sealed from it and providing a way to create other properties for unit testing.

**Comment by rtanase** [comment]
What about extracting an interface out of it and changing the IServicePartitionResolver to return an interface for the Resolve calls, instead of the concrete types? Do you guys have any internal dependency on that returned type being a ResolvedServicePartition instance?

**Comment by amanbha** [comment]
yes, that's correct, the concrete implementation is referenced at many places. I understand that having an interface or a public constructor would have been correct but its slightly older code base so need to be thought through to avoid breaking changes.
